web/src/components/profile/getErrorStateForAddressField.ts

```diff
@@ -96,24 +105,80 @@ export const getErrorStateForAddressField = ({
     });
   }
 
+  if (field === "addressState") {
+    return fieldWithAssociatedFields({
+      associatedFields: [
+        "addressLine1",
+        "addressCity",
+        "addressMunicipality",
+        "addressZipCode",
+        "addressLine2",
+      ],
+      label: (Config.formation.fields as any)[field].error,
+    });
+  }
+
+  if (field === "addressCity") {
+    return fieldWithAssociatedFields({
+      associatedFields: [
+        "addressLine1",
+        "addressProvince",
+        "addressCountry",
+        "addressZipCode",
+        "addressLine2",
+      ],
+      label: (Config.formation.fields as any)[field].error,
+    });
+  }
+
+  if (field === "addressProvince") {
+    return fieldWithAssociatedFields({
+      associatedFields: ["addressLine1", "addressCity", "addressCountry", "addressZipCode", "addressLine2"],
+      label: (Config.formation.fields as any)[field].error,
+    });
+  }
+
+  if (field === "addressCountry") {
+    return fieldWithAssociatedFields({
+      associatedFields: ["addressLine1", "addressCity", "addressProvince", "addressZipCode", "addressLine2"],
+      label: (Config.formation.fields as any)[field].error,
+    });
+  }
+
   if (field === "addressZipCode") {
-    const exists = !!addressData[field];
+    const exists = !!formationAddressData[field];
     let inRange = false;
+    const isValidUsZipCode = isZipCodeUs(formationAddressData[field]);
+    const isValidNjZipCode = isZipCodeNj(formationAddressData[field]);
+    const isIntlPostalCode = isZipCodeIntl(formationAddressData[field]);
+
+    if (formationAddressData.addressState?.shortCode === "NJ") {
+      inRange = isValidNjZipCode;
+    } else if (formationAddressData.addressState?.shortCode) {
+      inRange = isValidUsZipCode;
+    } else {
+      inRange = isIntlPostalCode;
+    }
```

Comment:
[dust] You're already tracking a var `inRange` that is getting set based on the result from each of these `isZipCodeX` functions. I don't think you need those additional vars. Personal preference would be to trim it down a bit.

```suggestion
    if (formationAddressData.addressState?.shortCode === "NJ") {
      inRange = isZipCodeNj(formationAddressData[field]);
    } else if (formationAddressData.addressState?.shortCode) {
      inRange = isZipCodeUs(formationAddressData[field]);
    } else {
      inRange = isZipCodeIntl(formationAddressData[field]);
    }
```